dapp/src/components/name-card/ExtendedNameCard.tsx

```diff
@@ -39,6 +40,28 @@ export function ExtendedNameCard({
     const menuOptions = getNameMenuOptions(nft, hasSubnames, openDialog, nameRecordData);
     const label = normalizeIotaName(nft.name);
 
+    const nameRecord = nameRecordData as
+        | Extract<NameRecordData, { type: 'unavailable' }>
+        | undefined;
+
+    const targetAddress = nameRecord?.nameRecord?.targetAddress;
+    const expired = nameRecord?.nameRecord ? isNameRecordExpired(nameRecord.nameRecord) : false;
+
+    const getButtonText = () => {
+        if (expired) {
+            return 'Renew Name';
+        }
+        return targetAddress ? formatAddress(targetAddress) : 'Connect to address';
+    };
+
+    const handleButtonClick = () => {
+        if (expired) {
+            openDialog(NameDialogId.RenewName);
+        } else if (!targetAddress) {
+            openDialog(NameDialogId.ConnectToAddress);
+        }
+    };
```

Comment:
we should open the dialog to manage (disconnect/set a target address)
```suggestion
    const handleButtonClick = () => {
        if (expired) {
            openDialog(NameDialogId.RenewName);
        } else {
            openDialog(NameDialogId.ConnectToAddress);
        }
    };
```